oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/inference/InferenceConfig.java

```diff
@@ -0,0 +1,226 @@
+/*
```

Comment:
would it make sense to move all these classes you have in this package in oak-search instead of oak-search-elastic?

Reply:
This is valid for elastic indexes only, so I think this is right place. We are also using  OSGI configs from oak-search-elastic